server/embed/config_logging.go

```diff
@@ -34,7 +34,7 @@ import (
 )
 
 // GetLogger returns the logger.
-func (cfg Config) GetLogger() *zap.Logger {
+func (cfg *Config) GetLogger() *zap.Logger {
 	cfg.loggerMu.RLock()
```

Comment:
This looks even as a bug. Locking mutex on value receiver means we are locking a copy of lock.

Reply:
It isn't a bug, the lock itself is a pointer.